### PR TITLE
fix: user secure random generator

### PR DESCRIPTION
**Pull request by ai**
`rndm` use `Math.random` which is predictable and non-secure. [`nanoid`](https://github.com/ai/nanoid) uses hardware random generator which is much faster.

Also `nanoid` is [very fast](https://github.com/ai/nanoid#benchmark) and [popular](https://www.npmjs.com/package/nanoid) tool.

### CODE REVIEW

**Review by fengmk2 — COMMENTED** [review]
nice job, I will take a look deeply and merge it today.

File: app/extend/context.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ module.exports = {
 
   get nonce() {
     if (!this._nonceCache) {
-      this._nonceCache = rndm(16);
+      this._nonceCache = nanoid(16);
```
Comment (fengmk2) [inline app/extend/context.js:45]: according benchmark result https://github.com/ai/nanoid#benchmark
nanoid will 10x slow then rndm(non-secure) version.

Comment (ai) [reply app/extend/context.js:45]: Yeap, this is a price for security. You can use `nanoid/non-secure`, but I think having real random numbers is more important than microoptimization here 😊

Comment (ghost) [reply app/extend/context.js:45]: @ai：If there's no obvious secure problem. I don't think it needs modifications, for 10x slower.

Comment (ai) [reply app/extend/context.js:45]: @Maledong good point :+1: If you want only performance I replaced `nanoid` with `nanoid/non-secure`, which is 2-3% faster than `rndm`. But I will understand if 2-3% of performance is not a big deal

(Of course, you can also remember that `nanoid` is a more active project and have more test, including statistic distribution tests, but it is not big deal as well).

Comment (Maledong) [reply app/extend/context.js]: Decision on @fengmk2 :D

**Comment by ai** [comment]
@fengmk2 Good news, everyone :).

1. My Chinese friends added [Chinese docs](https://github.com/ai/nanoid/blob/master/README-cn.md)
2. We [improved](https://github.com/ai/nanoid/commit/484ad4c3effdf352e3f9dbb5981d3e31c5a4d3b5) `nanoid/non-secure` performance and now we are 10% faster

```
$ ./test/benchmark
nanoid                354,201 ops/sec
nanoid/generate       348,467 ops/sec
uid.sync              325,347 ops/sec
uuid/v4               322,328 ops/sec
shortid                33,277 ops/sec

Async:
uid                    71,998 ops/sec
nanoid/async           72,836 ops/sec

Non-secure:
rndm                2,495,324 ops/sec
nanoid/non-secure   2,746,033 ops/sec
```

**Comment by fengmk2** [comment]
@ai thanks a lot!

**Review by fengmk2 — APPROVED** [review]

**Pull request merged**